### PR TITLE
Core/Player: Keep shared quests through map changes instead of getting stuck in busy state

### PR DESCRIPTION
**Pull request by Cannix**
**Changes proposed:**

Keep shared quests through map changes instead of getting stuck in busy state.
-  Send shared quest anew after map changes

**Target branch(es):** 3.3.5

- [X] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.)

- [x] Builds, tested in-game

### CODE REVIEW

**Comment by Cannix** [comment]
Oh, forgot about new commits directly added to a pull request. Haven't used pull requests in a while..

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 1]
```diff
@@ -375,6 +375,10 @@ class npc_tallhorn_stag : public CreatureScript
                 }
                 _phase = 0;
             }
+
```
Comment (Shauren) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: unrelated change?

**Comment by Shauren** [comment]
How does this behave on retail? Keep showing the dialog or just clear the state?

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp (hunk 1 above)
Comment (Cannix) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: Already wrote that this commit wasnt intended, shall i open a new pull request or just revert it?

Comment (Shauren) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: just revert, it wont be visible in history when we merge it

**Comment by Treeston** [comment]
@Cannix refactor as follows please:
`GetDividedQuest` -> `GetSharedQuestID`
`GetDivider` -> `GetPlayerSharingQuest`
`SetDivideInfo` -> `SetQuestSharingInfo`
`m_divider` -> `m_playerSharingQuest`
`m_dividedQuest` -> `m_sharedQuestId`
NEW: `ClearQuestSharingInfo` (set empty,0)

File: src/server/game/Handlers/QuestHandler.cpp [hunk 2]
```diff
@@ -117,7 +117,7 @@ void WorldSession::HandleQuestgiverAcceptQuestOpcode(WorldPacket& recvData)
 #define CLOSE_GOSSIP_CLEAR_DIVIDER() \
```
Comment (Treeston) [inline src/server/game/Handlers/QuestHandler.cpp]: Rename this macro while you're at it too, please (`CLOSE_GOSSIP_CLEAR_SHARING_INFO`)
